### DISCUSSION TITLE
Support for BETWEEN ... AND ...

### DISCUSSION THREAD

**Discussion by cptb** [bruce-dunwiddie/tsql-parser#130] [General]
Hi Bruce,
I saw an unhandled exception got thrown when parsing a statement contains BETWEEN ... AND...
E.g. 
`SELECT IIF(my_column BETWEEN 10 AND 20, 'A', 'B') FROM my_table;`

Error I got was: 
System.ArgumentOutOfRangeException: 'Non-negative number required.
Parameter name: count'

If change the query to be 
`SELECT IIF(my_column >=10 AND my_column <= 20, 'A', 'B') FROM my_table;`
Then it was parsed fine.

Is there an intention to add the support for BETWEEN?

Thank you very much.
Paul

**Comment by bruce-dunwiddie** [comment]
Thanks again for more findings.

There's intention to handle all valid T-SQL. I just haven't gotten around to some of the newer usage yet. Just very light on available time right now.

Thank you for the examples. I'll get to it eventually.

Bruce
